### PR TITLE
fix: now throwing a message collision error

### PR DESCRIPTION
**Pull request by krackerman**

### CODE REVIEW

File: src/module/sqs/send-message-batch.js [hunk 1]
```diff
@@ -20,6 +20,9 @@ const sendBatch = async (sqsBatch, queueUrl, {
     const id = objectHash(msg);
     const msgDelaySeconds = getDelaySeconds(msg);
     const delaySeconds = msgDelaySeconds === undefined ? batchDelaySeconds : msgDelaySeconds;
+    if (p[id] !== undefined) {
+      throw new MessageCollisionError(p[id]);
```
Comment (krackerman) [inline src/module/sqs/send-message-batch.js]: Do we want to pass back the whole message or just an id??

File: src/resources/errors.js [hunk 2]
```diff
@@ -1,8 +1,17 @@
...
 class SendMessageBatchError extends Error {
   constructor(message) {
     super(message);
     this.name = 'SendMessageBatchError';
   }
 }
 
+class MessageCollisionError extends Error {
+  constructor(message) {
+    super(JSON.stringify(message));
```
Comment (krackerman) [inline src/resources/errors.js]: We need to do this for the `SendMessageBatchError` error (`JSON.stringify`).

Comment (simlu) [reply src/resources/errors.js]: Why what?

**Pull request merged**